### PR TITLE
pmi config

### PR DESCRIPTION
**Pull request by raffenet**

### CODE REVIEW

**Comment by raffenet** [comment]
test:jenkins/ch4/ofi
test:jenkins/ch4/ucx
test:jenkins/ch3/tcp

**Comment by zhenggb72** [comment]
The first few commits are fine with me. The last commit to use PMI2, I am not sure. Changing CH4 to use PMI2 is a larger effort I thought Intel MPI was doing. They may have a patch for it.

**Comment by raffenet** [comment]
Okay. The reason I did that was because the PMI calls in shm init did not work when configuring with Cray PMI. If we go PMI2 across the board, everything just worked. Maybe @ddurnov can comment on any PMI2 patch for CH4 from Intel?

**Comment by raffenet** [comment]
It also occurs to me that in CH4, the netmods are calling `PMI_Finalize`, but that should move to the CH4 level. That's where `PMI_Init` happens after all. I will add another patch for that.

File: src/mpid/ch4/netmod/ofi/ofi_init.h [hunk 1]
```diff
@@ -755,7 +755,8 @@ static inline int MPIDI_NM_mpi_init_hook(int rank,
             sprintf(keyS, "OFI-%d", i);
 #ifdef USE_CRAYPMI_API
             MPIDI_OFI_PMI_CALL_POP(PMI2_KVS_Get
-                                   (NULL, -1, keyS, valS, MPIDI_KVSAPPSTRLEN, &len), pmi);
```
Comment (zhenggb72) [inline src/mpid/ch4/netmod/ofi/ofi_init.h]: the int len couple lines above should also be removed?

**Comment by raffenet** [comment]
test:jenkins/ch4/ofi
test:jenkins/ch4/ucx
test:jenkins/ch3/tcp

**Pull request merged**